pkg/server/server.go

```diff
@@ -383,31 +386,32 @@ func (s *Server) VerifyEmail(c *gin.Context, redirectURL string) {
 	if !exists {
 		s.redirectError(c, errors.ErrInvalidAccessToken)
 		// Redirecting user to UI if the user is not authorized
-		c.Redirect(http.StatusUnauthorized, redirectURL)
+		c.Redirect(http.StatusPermanentRedirect, redirectURL)
 		return
 	}
 	jwtUserCredentials := jwtUser.(*models.UserCredentials)
 
 	if jwtUserCredentials.UnverifiedEmail != nil {
-		*jwtUserCredentials.Email = jwtUserCredentials.GetUnverifiedEmail()
-		*jwtUserCredentials.UnverifiedEmail = ""
+		tmp := jwtUserCredentials.GetUnverifiedEmail()
+		jwtUserCredentials.Email = &tmp
+		jwtUserCredentials.UnverifiedEmail = nil
 	} else {
 		log.Errorln("No email found to be verified for user uid: ", jwtUserCredentials.GetUID())
 		c.JSON(http.StatusBadRequest, gin.H{
 			"error": "No email found to be verified",
 		})
 		// Redirecting user to UI if no email found in field `unverified_email`
-		c.Redirect(http.StatusBadRequest, redirectURL)
+		c.Redirect(http.StatusPermanentRedirect, redirectURL)
 		return
 	}
 
 	_, err := usermanager.UpdateUserDetails(s.userStore, jwtUserCredentials)
 	if err != nil {
 		s.redirectError(c, err)
 		// Redirecting user to UI if updating the database fails
-		c.Redirect(http.StatusInternalServerError, redirectURL)
+		c.Redirect(http.StatusPermanentRedirect, redirectURL)
```

Comment:
Why change the error code?

Reply:
Same comment as https://github.com/mayadata-io/kubera-auth/pull/45/files#r592392465